src/OrbitQt/HistogramWidget.h

```diff
@@ -19,15 +24,33 @@ class HistogramWidget : public QWidget {
  public:
   using QWidget::QWidget;
 
-  void UpdateHistogram(std::optional<orbit_statistics::Histogram> histogram,
-                       std::string function_name);
+  void UpdateData(std::vector<uint64_t> data, std::string function_name);
 
  protected:
   void paintEvent(QPaintEvent* /*event*/) override;
 
+  void mousePressEvent(QMouseEvent* event) override;
+  void mouseReleaseEvent(QMouseEvent* event) override;
+  void mouseMoveEvent(QMouseEvent* event) override;
+
  private:
-  std::optional<orbit_statistics::Histogram> histogram_;
-  std::optional<std::string> function_name_;
+  [[nodiscard]] bool IsSelectionActive() const;
+
+  [[nodiscard]] uint64_t MinValue() const;
+  [[nodiscard]] uint64_t MaxValue() const;
+
+  [[nodiscard]] int Width() const;
+  [[nodiscard]] int Height() const;
+  [[nodiscard]] int WidthMargin() const;
+  [[nodiscard]] int HeightMargin() const;
+
+  std::optional<const std::vector<uint64_t>> data_;
+  std::optional<const std::string> function_name_;
```

Comment:
It looks to me if these two members share their lifetime. What about putting them into a struct and putting the struct into an optional?

Reply:
And one more thing (in this case rather minor - so feel free to ignore): Making these members `const` means the whole class can't be moved/swapped/etc. As I said not very relevant here because the class is a widget. Nevertheless it's not good practice. The C++ way of enforcing these members to be non-mutable (after setting them once) is to make them private and offer const getters. Since these members are already private and you wanna guard against mutation from other parts in the code, the proper way would be to put them into a separate class and enforce non-mutability there by making them private and only offering getters.